src/App.tsx

```diff
@@ -15,36 +16,86 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortType {
+  NONE = '',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.NONE);
+  const [reverseField, setReverseField] = useState(false);
+
+  const sortedGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  const visibleGoods = reverseField ? [...sortedGoods].reverse() : sortedGoods;
+
+  const reset = () => {
+    setSortField(SortType.NONE);
+    setReverseField(false);
+  };
+
+  const showResetButton = sortField !== SortType.NONE || reverseField;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+          onClick={() => setSortField(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+          onClick={() => setSortField(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !reverseField,
+          })}
+          onClick={() => {setReverseField(!reverseField)}}
```

Comment:
```suggestion
          onClick={() => {setReverseField(isReversed => !isReversed)}}
```
Always use callback when update state based on previous value